ThirdHW/ThirdHW/MyTask.cs

```diff
@@ -0,0 +1,116 @@
+﻿namespace ThirdHW;
+
+public class MyTask<T1> : IMyTask<T1>
+{
+	private bool isCompleted;
+
+	private CancellationToken cancellationToken;
+
+	private MyThreadPool myThreadPool;
+
+	private Func<T1> function;
+
+	private T1? result;
+
+	private Exception? exception;
+
+	private ManualResetEvent manualResetEventForContinuations;
+
+	private ManualResetEvent manualResetEventForResult;
+
+	private ManualResetEvent? manualResetEventIsUpperTaskCompleted;
+
+	private List<Action> continuations;
+
+	public MyTask(Func<T1> func, MyThreadPool myThreadPool,
+		CancellationToken cancellationToken,
+		ManualResetEvent? manualResetEventIsUpperTaskCompleted = null)
+	{
+		function = func;
+		this.myThreadPool = myThreadPool;
+		manualResetEventForContinuations = new ManualResetEvent(false);
+		manualResetEventForResult = new ManualResetEvent(false);
+		this.manualResetEventIsUpperTaskCompleted = manualResetEventIsUpperTaskCompleted;
+		IsContinuation = manualResetEventIsUpperTaskCompleted == null ? false :
+			true;
+		continuations = new List<Action>();
+	}
+
+	public bool IsContinuation { get; }
+
+	public bool IsCompleted { get => isCompleted; }
+
+	public T1 Result
+	{
+		get
+		{
+			if (cancellationToken.IsCancellationRequested && !isCompleted)
+			{
+				throw new InvalidOperationException("shutdown was requested" +
+					"and task hadn't been completed");
+			}
+			manualResetEventForResult.WaitOne();
+			if (exception != null)
+			{
+				throw new AggregateException(exception);
+			} else
+			{
```

Comment:
Здесь и в других местах
```suggestion
			}
			else
			{
```